File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& a, int k) {
        unordered_map<int,int> mp; int sum=0; int ans=0;
        mp[0]=1; 
        for(int i=0;i<a.size();i++)
        {
            sum+=a[i];
            if(mp.find(sum-k)!=mp.end())
            {
                ans+=mp[sum-k];
            }
            mp[sum]++;
        }
        return ans;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
       HashMap<Integer,Integer> mp=new HashMap<>(); 
        mp.put(0,1);
        int cnt=0; int sum=0;
        for(int i=0;i<nums.length;i++)
        {
            sum+=nums[i];
            if(mp.containsKey(sum-k))
            {
                cnt+=mp.get(sum-k);
            }
            mp.put(sum,mp.getOrDefault(sum,0)+1);
        }
        return cnt;
    }
}